File: php/features.php
<?php
declare(strict_types=1);

// Hook0 SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class Hook0Features
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new Hook0BaseFeature();
            case "test":
                return new Hook0TestFeature();
            default:
                return new Hook0BaseFeature();
        }
    }
}

File: py/features.py
# Hook0 SDK feature factory

from feature.base_feature import Hook0BaseFeature
from feature.test_feature import Hook0TestFeature


def _make_feature(name):
    features = {
        "base": lambda: Hook0BaseFeature(),
        "test": lambda: Hook0TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
